### PR TITLE
Parser Fix for LineNumberLovers

### LINKED ISSUE #3216: Parse Errors seem to be associated with line numbers on IF blocks
Not sure if this is related to issue #3012

This code Fails to parse:
```vb
Sub TestRubberduckParsing()
100       If True Then
102           Debug.Print "True"
104       Else
106           Debug.Print "False"
108       End If
End Sub
```

Also fails to parse:
```vb
Sub TestRubberduckParsing()
100       If True Then
           Debug.Print "True"
102       Else
           Debug.Print "False"
106       End If
End Sub
```

This code Does parse:
```vb
Sub TestRubberduckParsing()
100       If True Then
              Debug.Print "True"
          Else
              Debug.Print "False"
          End If
End Sub
```

This code will also parse:
```vb
Sub TestRubberduckParsing()
       If True Then
102           Debug.Print "True"
       Else
106           Debug.Print "False"
       End If
End Sub
```

This code will also parse:
```vb
Sub TestRubberduckParsing()
100       If True Then
102           Debug.Print "True"
       Else
106           Debug.Print "False"
       End If
End Sub
```

I think this also impacts Do While ... and some other blocks but have not had the chance to check properly.

### PR DESCRIPTION
**Pull request by retailcoder**
closes #3216.

### CODE REVIEW

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 1]
```diff
@@ -303,15 +303,15 @@ lastLetter : unrestrictedIdentifier;
 doLoopStmt :
```
Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: @MDoerner thoughts?

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 2]
```diff
@@ -467,9 +467,9 @@ nameStmt : NAME whiteSpace expression whiteSpace AS whiteSpace expression;
 // 5.4.2.10 Select Case Statement
 selectCaseStmt :
     SELECT whiteSpace? CASE whiteSpace? selectExpression endOfStatement
-    caseClause*
-    caseElseClause?
-    END_SELECT
+    (statementLabelDefinition? whiteSpace? caseClause)*
+    statementLabelDefinition? whiteSpace? caseElseClause?
```
Comment (MDoerner) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: Does thin interfere with the quickfixes for the empty if block and empty else block inspections?

Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: hmm, very good question. needs a test.

Comment (retailcoder) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: It shouldn't, actually - these aren't counted as executable ststements. Need to check if it interferes with the quickfix though...

Comment (MDoerner) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: Hm, I put the comment at the wrong spot.

File: Rubberduck.Parsing/Grammar/VBAParser.g4 [hunk 3]
```diff
@@ -362,11 +362,11 @@ goToStmt : GOTO whiteSpace expression;
 ifStmt :
      IF whiteSpace booleanExpression whiteSpace THEN endOfStatement
      block
-     elseIfBlock*
-     elseBlock?
-     END_IF
+     (statementLabelDefinition? whiteSpace? elseIfBlock)*
+     (statementLabelDefinition? whiteSpace? elseBlock?)
+     statementLabelDefinition? whiteSpace? END_IF
```
Comment (MDoerner) [inline Rubberduck.Parsing/Grammar/VBAParser.g4]: This is where my first comment should have gone. The problem I see here is that the line number is not part of the elseif block or else block. That might cause problems with the quickfixes.

**Pull request merged**